File: task_equivalents/llm_translations/llama/optimized/python/statistical_modelling_analysis/01_linear_regression_inference.py
import pandas as pd
import statsmodels.api as sm

# Load data
auto = pd.read_csv('data/csv/Auto.csv', na_values='?').dropna()

# Fit the model
X = sm.add_constant(auto['horsepower'])
model1 = sm.OLS(auto['mpg'], X).fit()

# Print summary
print(model1.summary())

# Make predictions
new_data = pd.DataFrame({'horsepower': [98]})
new_data = sm.add_constant(new_data)

# Confidence interval
confidence_interval = model1.get_prediction(new_data).summary_frame(alpha=0.05)[['mean', 'mean_ci_lower', 'mean_ci_upper']]
print(confidence_interval)

# Prediction interval
prediction_interval = model1.get_prediction(new_data).summary_frame(alpha=0.05)[['mean', 'obs_ci_lower', 'obs_ci_upper']]
print(prediction_interval)

File: task_equivalents/llm_translations/llama/optimized/r/statistical_modelling_analysis/01_linear_regression_inference.R
library(readr)
library(broom)

Auto <- read_csv('data/csv/Auto.csv', na = c("?", "NA")) %>%
  drop_na()

X <- cbind(1, Auto$horsepower)
Y <- Auto$mpg

model1_fit <- lm(Y ~ X[, 2])
summary_model <- tidy(summary(model1_fit))
print(summary(model1_fit))

xnew <- data.frame(horsepower = 98)
pred <- predict(model1_fit, newdata = xnew, interval = "confidence", level = 0.95)
print(data.frame(mean = pred[, 1], mean_ci_lower = pred[, 2], mean_ci_upper = pred[, 3]))

pred <- predict(model1_fit, newdata = xnew, interval = "prediction", level = 0.95)
print(data.frame(mean = pred[, 1], obs_ci_lower = pred[, 2], obs_ci_upper = pred[, 3]))